src/WebJobs.Script/Host/ScriptHostManager.cs

```diff
@@ -332,6 +332,9 @@ private static void LogErrors(ScriptHost host)
         /// <param name="forceStop">Forces the call to stop and dispose of the instance, even if it isn't present in the live instances collection.</param>
         private async Task Orphan(ScriptHost instance, bool forceStop = false)
         {
+            // Dispose primary host coordinator immediately so only new instance will have primary lease
+            instance.DisposePrimaryHostCoordinator();
```

Comment:
This concern should be encapsulated in the stop/disposal logic. Can we just handle the disposal of the primary host coordinator when we make the stop call and avoid exposing this public method? That shouldn't delay this too much.

Reply:
The Orphan call is fired off in a Task.Run and the Stop is embedded in there, so we wouldn't be able to guarantee that it actually happened before the new host is constructed

Reply:
This is very similar to some of these other lease-renewal issues we've had when using the InstanceId as the lease id -- you assume there's only one-at-a-time, but during these loops, you can end up with two that are renewing/releasing the lock unexpectedly.

Since Orphan is run in a Task, there's still no guarantee that this is complete before the next host starts, is there?